// retell-sdk/resources/call.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Call = void 0;
const resource_1 = require("../resource.js");
class Call extends resource_1.APIResource {
    /**
     * Retrieve details of a specific call
     *
     * @example
     * ```ts
     * const callResponse = await client.call.retrieve(
     *   '119c3f8e47135a29e65947eeb34cf12d',
     * );
     * ```
     */
    retrieve(callId, options) {
        return this._client.get(`/v2/get-call/${callId}`, options);
    }
    /**
     * Update metadata and sensitive data storage settings for an existing call.
     *
     * @example
     * ```ts
     * const callResponse = await client.call.update(
     *   'call_a4441234567890777c4a4a123e6',
     *   {
     *     metadata: {
     *       customer_id: 'cust_123',
     *       notes: 'Follow-up required',
     *     },
     *     opt_out_sensitive_data_storage: true,
     *   },
     * );
     * ```
     */
    update(callId, body, options) {
        return this._client.patch(`/v2/update-call/${callId}`, { body, ...options });
    }
    /**
     * Retrieve call details
     *
     * @example
     * ```ts
     * const callResponses = await client.call.list();
     * ```
     */
    list(body, options) {
        return this._client.post('/v2/list-calls', { body, ...options });
    }
    /**
     * Delete a specific call and its associated data
     *
     * @example
     * ```ts
     * await client.call.delete(
     *   '119c3f8e47135a29e65947eeb34cf12d',
     * );
     * ```
     */
    delete(callId, options) {
        return this._client.delete(`/v2/delete-call/${callId}`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Create a new outbound phone call
     *
     * @example
     * ```ts
     * const phoneCallResponse = await client.call.createPhoneCall(
     *   {
     *     from_number: '+14157774444',
     *     to_number: '+12137774445',
     *   },
     * );
     * ```
     */
    createPhoneCall(body, options) {
        return this._client.post('/v2/create-phone-call', { body, ...options });
    }
    /**
     * Create a new web call
     *
     * @example
     * ```ts
     * const webCallResponse = await client.call.createWebCall({
     *   agent_id: 'oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD',
     * });
     * ```
     */
    createWebCall(body, options) {
        return this._client.post('/v2/create-web-call', { body, ...options });
    }
    /**
     * Register a new phone call for custom telephony
     *
     * @example
     * ```ts
     * const phoneCallResponse =
     *   await client.call.registerPhoneCall({
     *     agent_id: 'oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD',
     *   });
     * ```
     */
    registerPhoneCall(body, options) {
        return this._client.post('/v2/register-phone-call', { body, ...options });
    }
}
exports.Call = Call;
