// retell-sdk/resources/chat.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../resource.mjs";
export class Chat extends APIResource {
    /**
     * Create a chat session
     *
     * @example
     * ```ts
     * const chatResponse = await client.chat.create({
     *   agent_id: 'oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/create-chat', { body, ...options });
    }
    /**
     * Retrieve details of a specific chat
     *
     * @example
     * ```ts
     * const chatResponse = await client.chat.retrieve(
     *   '16b980523634a6dc504898cda492e939',
     * );
     * ```
     */
    retrieve(chatId, options) {
        return this._client.get(`/get-chat/${chatId}`, options);
    }
    /**
     * List all chats
     *
     * @example
     * ```ts
     * const chatResponses = await client.chat.list();
     * ```
     */
    list(options) {
        return this._client.get('/list-chat', options);
    }
    /**
     * Create a chat completion message
     *
     * @example
     * ```ts
     * const response = await client.chat.createChatCompletion({
     *   chat_id: 'oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD',
     *   content: 'hi how are you doing?',
     * });
     * ```
     */
    createChatCompletion(body, options) {
        return this._client.post('/create-chat-completion', { body, ...options });
    }
    /**
     * End an ongoing chat
     *
     * @example
     * ```ts
     * await client.chat.end('16b980523634a6dc504898cda492e939');
     * ```
     */
    end(chatId, options) {
        return this._client.patch(`/end-chat/${chatId}`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
}
